src/openrct2/scripting/bindings/entity/ScGuest.cpp

```diff
@@ -494,6 +499,291 @@ namespace OpenRCT2::Scripting
+        {
+            return false;
+        }
+
+        if (*shopItem == ShopItem::Voucher)
+        {
+            if (item["voucherType"].type() == DukValue::Type::STRING)
+            {
+                // Voucher
+                auto voucher = VoucherTypeMap.TryGet(item["voucherType"].as_string());
+                if (!voucher || *voucher != peep->VoucherType)
+                {
+                    return false;
+                }
+
+                if (*voucher == VOUCHER_TYPE_RIDE_FREE)
+                {
+                    if (item["rideId"].type() == DukValue::Type::NUMBER)
+                    {
+                        // RideVoucher
+                        if (item["rideId"].as_uint() != peep->VoucherRideId.ToUnderlying())
+                        {
+                            return false;
+                        }
+                    }
+                }
+                else if (*voucher == VOUCHER_TYPE_FOOD_OR_DRINK_FREE)
+                {
+                    if (item["item"].type() == DukValue::Type::STRING)
+                    {
+                        // FoodDrinkVoucher
+                        auto voucherItem = ShopItemMap.TryGet(item["item"].as_string());
+                        if (!voucherItem || *voucherItem != peep->VoucherShopItem)
+                        {
+                            return false;
+                        }
+                    }
+                }
+            }
+        }
+        else if (GetShopItemDescriptor(*shopItem).IsPhoto())
+        {
+            // GuestPhoto
+            if (item["rideId"].type() == DukValue::Type::NUMBER)
+            {
+                switch (*shopItem)
+                {
+                    case ShopItem::Photo:
+                        if (item["rideId"].as_uint() != peep->Photo1RideRef.ToUnderlying())
+                        {
+                            return false;
+                        }
+                        break;
+                    case ShopItem::Photo2:
+                        if (item["rideId"].as_uint() != peep->Photo2RideRef.ToUnderlying())
+                        {
+                            return false;
+                        }
+                        break;
+                    case ShopItem::Photo3:
+                        if (item["rideId"].as_uint() != peep->Photo3RideRef.ToUnderlying())
+                        {
+                            return false;
+                        }
+                        break;
+                    case ShopItem::Photo4:
+                        if (item["rideId"].as_uint() != peep->Photo4RideRef.ToUnderlying())
+                        {
+                            return false;
+                        }
+                        break;
+                    default:
+                        return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    void ScGuest::give_item(const DukValue& item) const
```

Comment:
Wouldn't it be better if this function would return some errors if the input is invalid, instead of not doing anything?

Reply:
That makes sense. I added error messages to describe how the item passed is invalid.